apollo-audit/apollo-audit-impl/src/test/java/com/ctrip/framework/apollo/audit/aop/ApolloAuditSpanAspectTest.java

```diff
@@ -0,0 +1,143 @@
+    verify(api, times(1))
+        .appendAuditLog(eq(opType), eq(opName), eq(description));
+    verify(mockScope, times(1))
+        .close();
+    verify(aspect, times(1))
+        .auditDataInfluenceArg(eq(mockPJP));
+  }
+
+  @Test
+  public void testAuditDataInfluenceArg() throws NoSuchMethodException {
+    ProceedingJoinPoint mockPJP = mock(ProceedingJoinPoint.class);
+    Object[] args = new Object[]{new Object(), new Object()};
+    Method method = MockAuditClass.class.getMethod("mockAuditMethod", Object.class, Object.class);
+    {
+      doReturn(method).when(aspect).findMethod(any());
+      when(mockPJP.getArgs()).thenReturn(args);
+    }
+    aspect.auditDataInfluenceArg(mockPJP);
+    verify(aspect, times(1))
+        .parseArgAndAppend(eq("App"), eq("Name"), eq(args[0]));
+  }
+
+  @Test
+  public void testFindMethod() throws NoSuchMethodException {
+    ProceedingJoinPoint mockPJP = mock(ProceedingJoinPoint.class);
+    MockAuditClass mockAuditClass = new MockAuditClass();
+    Signature signature = mock(Signature.class);
+    Method method = MockAuditClass.class.getMethod("mockAuditMethod", Object.class, Object.class);
+    {
+      when(mockPJP.getTarget()).thenReturn(mockAuditClass);
+      when(mockPJP.getSignature()).thenReturn(signature);
+      when(signature.getName()).thenReturn("mockAuditMethod");
+    }
+    Method methodFounded = aspect.findMethod(mockPJP);
+
+    assertEquals(method, methodFounded);
+  }
+
+  @Test
+  public void testParseArgAndAppendCaseNullName() {
+    Object somewhat = new Object();
+    aspect.parseArgAndAppend(null, null, somewhat);
+    verify(api, times(0))
+        .appendDataInfluence(any(), any(), any(), any());
+  }
+
+  @Test
+  public void testParseArgAndAppendCaseCollectionTypeArg() {
+    final String entityName = "App";
+    final String fieldName = "Name";
+    List<Object> list = Arrays.asList(new Object(), new Object(), new Object());
+
+    {
+      doNothing().when(api).appendDataInfluence(any(), any(), any(), any());
+    }
+    aspect.parseArgAndAppend(entityName, fieldName, list);
+    verify(api, times(list.size())).appendDataInfluence(eq(entityName),
+        eq(ApolloAuditConstants.ANY_MATCHED_ID), eq(fieldName), any());
+  }
+
+  @Test
+  public void testParseArgAndAppendCaseNormalTypeArg() {
+    final String entityName = "App";
+    final String fieldName = "Name";
+    Object arg = new Object();
+
+    {
+      doNothing().when(api).appendDataInfluence(any(), any(), any(), any());
+    }
+    aspect.parseArgAndAppend(entityName, fieldName, arg);
+    verify(api, times(1)).appendDataInfluence(eq(entityName),
+        eq(ApolloAuditConstants.ANY_MATCHED_ID), eq(fieldName), any());
+  }
+
+  public class MockAuditClass {
+
+    public void mockAuditMethod(
+        @ApolloAuditLogDataInfluence
+        @ApolloAuditLogDataInfluenceTable(tableName = "App")
+        @ApolloAuditLogDataInfluenceTableField(fieldName = "Name") Object val1,
```

Comment:
add a test case when arg is a collection?

Reply:
> add a test case when arg is a collection?

There is a method to parsing arg, and a test method too, see `testParseArgAndAppendCaseCollectionTypeArg` at ApolloAuditSpanAspectTest